src/main/java/com/dev/cinema/dao/impl/UserDaoImpl.java

```diff
@@ -0,0 +1,56 @@
+package com.dev.cinema.dao.impl;
+
+import com.dev.cinema.dao.UserDao;
+import com.dev.cinema.exceptions.DataBindingException;
+import com.dev.cinema.lib.Dao;
+import com.dev.cinema.model.User;
+import com.dev.cinema.util.HashUtil;
+import com.dev.cinema.util.HibernateUtil;
+import java.util.Optional;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class UserDaoImpl implements UserDao {
+    @Override
+    public User add(User user) {
+        byte [] salt = HashUtil.getSalt();
+        user.setPassword(HashUtil.getHashPassword(user
+                .getPassword(), salt));
+        user.setSalt(salt);
```

Comment:
This should be done in [UserService](https://mate-academy.github.io/jv-program-fulltime/03_hibernate/checklist/hibernate_05.html). It's not a part of DAO